homeassistant/components/envisalink/__init__.py

```diff
@@ -67,35 +68,41 @@
 
 CONFIG_SCHEMA = vol.Schema(
     {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_HOST): cv.string,
-                vol.Required(CONF_PANEL_TYPE): vol.All(
-                    cv.string, vol.In([PANEL_TYPE_HONEYWELL, PANEL_TYPE_DSC])
-                ),
-                vol.Required(CONF_USERNAME): cv.string,
-                vol.Required(CONF_PASS): cv.string,
-                vol.Optional(CONF_CODE): cv.string,
-                vol.Optional(CONF_PANIC, default=DEFAULT_PANIC): cv.string,
-                vol.Optional(CONF_ZONES): {vol.Coerce(int): ZONE_SCHEMA},
-                vol.Optional(CONF_PARTITIONS): {vol.Coerce(int): PARTITION_SCHEMA},
-                vol.Optional(CONF_EVL_PORT, default=DEFAULT_PORT): cv.port,
-                vol.Optional(CONF_EVL_VERSION, default=DEFAULT_EVL_VERSION): vol.All(
-                    vol.Coerce(int), vol.Range(min=3, max=4)
-                ),
-                vol.Optional(CONF_EVL_KEEPALIVE, default=DEFAULT_KEEPALIVE): vol.All(
-                    vol.Coerce(int), vol.Range(min=15)
-                ),
-                vol.Optional(
-                    CONF_ZONEDUMP_INTERVAL, default=DEFAULT_ZONEDUMP_INTERVAL
-                ): vol.Coerce(int),
-                vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): vol.Coerce(int),
-            }
+        DOMAIN: vol.All(
+            # Deprecated in 2024.7, remove from config in 2025.1
+            cv.deprecated(CONF_CODE),
+            vol.Schema(
+                {
+                    vol.Required(CONF_HOST): cv.string,
+                    vol.Required(CONF_PANEL_TYPE): vol.All(
+                        cv.string, vol.In([PANEL_TYPE_HONEYWELL, PANEL_TYPE_DSC])
+                    ),
+                    vol.Required(CONF_USERNAME): cv.string,
+                    vol.Required(CONF_PASS): cv.string,
+                    vol.Optional(CONF_CODE): cv.string,
+                    vol.Optional(CONF_PANIC, default=DEFAULT_PANIC): cv.string,
+                    vol.Optional(CONF_ZONES): {vol.Coerce(int): ZONE_SCHEMA},
+                    vol.Optional(CONF_PARTITIONS): {vol.Coerce(int): PARTITION_SCHEMA},
+                    vol.Optional(CONF_EVL_PORT, default=DEFAULT_PORT): cv.port,
+                    vol.Optional(
+                        CONF_EVL_VERSION, default=DEFAULT_EVL_VERSION
+                    ): vol.All(vol.Coerce(int), vol.Range(min=3, max=4)),
+                    vol.Optional(
+                        CONF_EVL_KEEPALIVE, default=DEFAULT_KEEPALIVE
+                    ): vol.All(vol.Coerce(int), vol.Range(min=15)),
+                    vol.Optional(
+                        CONF_ZONEDUMP_INTERVAL, default=DEFAULT_ZONEDUMP_INTERVAL
+                    ): vol.Coerce(int),
+                    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): vol.Coerce(
+                        int
+                    ),
+                    vol.Optional(CONF_UNIQUE_ID): cv.string,
```

Comment:
We no longer allow integrations to add or change a platform or integration YAML configuration.

More information on this can be found in:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

Reply:
Hmm. Wasn't adding a unique id an exception to that rule?

Reply:
Only if the integration isn't migratable under ADR-0010; this one is.